library/alloc/src/sync.rs

```diff
@@ -1357,11 +1357,13 @@ impl<T: ?Sized> Clone for Arc<T> {
 
         // However we need to guard against massive refcounts in case someone
         // is `mem::forget`ing Arcs. If we don't do this the count can overflow
-        // and users will use-after free. We racily saturate to `isize::MAX` on
-        // the assumption that there aren't ~2 billion threads incrementing
-        // the reference count at once. This branch will never be taken in
+        // and users will use-after free. This branch will never be taken in
         // any realistic program.
         //
+        // This check is not 100% water-proof: we error when the refcount grows beyond `isize::MAX`
+        // on the assumption that there aren't ~2 billion threads incrementing the reference count
```

Comment:
Maybe worth noting where the 2 billion comes from? On a 16-bit platform for example this is much more of a problem (not that we support those today, I think?).

Reply:
So no test that triggers this logic on 64 bit pointer systems ? :)
This logic Is more a bout detecting corrupted memory.
Could this functionality be configurable ?.  So can get smaller code size, and 1 less branch.